### PR TITLE
[FIX] account, l10n_ar_withholding: fix install with multiple AR companies

### PR DESCRIPTION
**Pull request by kitan191**
Steps to reproduce:
- Install Accounting and l10n_ar
- Create 2 companies located in Argentina (e.g. Company A & Company B)
- Switch to Company A
- Go to Accounting settings
- Set Fiscal Localization to Argentina (e.g. Argentina - Argentine Generic Chart of Accounts for Excempt individuals)
- Switch to Company B
- Go to Accounting settings
- Set Fiscal Localization to the same Fiscal Localization than Company A (i.e. Argentina - Argentine Generic Chart of Accounts for Excempt individuals)
- Try to install "l10n_ar_withholding" module

Issue:
A User Error is raised:
"Incompatible companies on records:
- 'account.tax.repartition.line,1230' belongs to company 'Company A' and 'Account' (account_id: '1.1.4.03.020 SUSS Withholding incurred') belongs to another company.
- 'account.tax.repartition.line,1232' belongs to company 'Company A' and 'Account' (account_id: '1.1.4.03.020 SUSS Withholding incurred') belongs to another company.
- 'account.tax.repartition.line,1234' belongs to company 'Company A' and 'Account' (account_id: '1.1.4.05.030 Withholdings of Profits incurred') belongs to another company.
- 'account.tax.repartition.line,1236' belongs to company 'Company A' and 'Account' (account_id: '1.1.4.05.030 Withholdings of Profits incurred') belongs to another company."

Cause:
When installing "l10n_ar_withholding" module, each Argentine company is updated with some tax data.
These data are "generic" (not linked to any company) and used for each company, but some treatment is performed on them by the first Argentine company, linking them to the account ids of that company.
The following companies are then updated with data linked to the first company.

Solution:
Compute the data for each company.

opw-3709819

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by clbr-odoo — COMMENTED** [review]
Hello :wave: 

So, I like your fix because at the moment we are making disk read with `_parse_csv` even if there are no company at all using one of the AR templates.
But, I also don't like it because we are still making multiple `parse` per company.
I think the underlying problem is that `_load_data(data)` is modifying the `data` parameter... I would fix it in `account` module by adding a `deepcopy` at the beginning of the method. Then change a bit your fix so that the parse is only called once per template_code that is actually used by one of the companies.

What do you think ? :thinking:

**Comment by kitan191** [comment]
Hi @clbr-odoo 

Thanks for the review. :+1: 

Yes. I agree with you. :wink: 
I've added a `deepcopy` in `_load_data` method at the only place where `data` is used.

File: addons/l10n_ar_withholding/__init__.py
Comment (clbr-odoo) [inline addons/l10n_ar_withholding/__init__.py:1]:
```python
def _l10n_ar_withholding_post_init(env):
    template_codes = ('ar_ri', 'ar_ex', 'ar_base')
    ar_companies = env['res.company'].search([('chart_template', 'in', template_codes)])
    used_template_codes = set(ar_companies.mapped('chart_template'))
    for used_template_code in used_template_codes:
        data = {... _parse_csv(...)}
        for company in ar_companies.filtered(lambda c: c.chart_template == used_template):
            # all the stuff in the company loop 
        
```
Something like this (not tested) would avoid the possibly useless `_parse_csv(...)` calls :thinking: 
At the moment it's called 3 times even if no company at all use the AR chart templates, which is a bit meh.

**Review by clbr-odoo — CHANGES_REQUESTED** [review]
Exactly that for the `account` modifications, but you lost the nice improvements you made for the `l10n_ar_withholding` side :sweat_smile:

File: addons/l10n_ar_withholding/__init__.py
Comment (kitan191) [reply addons/l10n_ar_withholding/__init__.py:1]: Yes. Indeed. I have reverted by changes too quickly :sweat_smile: 

Thanks 😊

Comment (clbr-odoo) [reply addons/l10n_ar_withholding/__init__.py:1]: Perfect thanks ! Sending it for final review :+1:

**Comment by csnauwaert** [comment]
@robodoo r+

**Pull request closed without merging**